### PR TITLE
Update usage details in the UCCSD template

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
The PR updates the code block in the usage details of the UCCSD template.

**Description of the Change:**
- An updated code example is added to the usage details of `qml.UCCSD`.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/templates/subroutines/uccsd.py [hunk 1]
```diff
@@ -104,38 +104,52 @@ class UCCSD(Operation):
...
+            optimizer = qml.GradientDescentOptimizer(stepsize=0.5)
 
-            # Compute the expectation value of 'h' for given set of parameters 'params'
-            params = np.random.normal(0, np.pi, len(singles) + len(doubles))
-            print(cost_fn(params))
+            # Optimize the circuit parameters and compute the energy
+            for n in range(20):
+                params, energy = optimizer.step_and_cost(circuit, params,
+                wires=range(qubits), s_wires=s_wires, d_wires=d_wires, hf_state=hf_state)
+                print("step = {:},  E = {:.8f} H".format(n, energy))
```
Comment (obliviateandsurrender) [inline pennylane/templates/subroutines/uccsd.py:152]: Do we show the output in such cases generally?

File: pennylane/templates/subroutines/uccsd.py [hunk 2]
```diff
@@ -104,38 +104,52 @@ class UCCSD(Operation):
...
             # Define the device
-            dev = qml.device('default.qubit', wires=qubits)
+            dev = qml.device("default.qubit", wires=qubits)
+
+            # Define the qnode
+            @qml.qnode(dev)
+            def circuit(params, wires, s_wires, d_wires, hf_state):
+
+                # Flip the HF state
+                hf_state = np.flip(hf_state)
```
[the remark this answers is not in the document]
Comment (soranjh) [reply pennylane/templates/subroutines/uccsd.py:137]: Because of this line [here](https://github.com/PennyLaneAI/pennylane/blob/07140375176c0791a50beac5e9aa0d9159026f8b/pennylane/templates/subroutines/uccsd.py#L207):

`init_state_flipped = np.flip(init_state)`

We can remove that line and not flip the HF state here. Not sure if it affects the accuracy in other workflows or not. But the state needs to be flipped to get correct energies for everything beyond `H2/sto-3g`.

[the remark this answers is not in the document]
Comment (soranjh) [reply pennylane/templates/subroutines/uccsd.py]: Removed the redundant flips of the initial state.

**Review by obliviateandsurrender — APPROVED** [review]
Looking good @soranjh! Let's fix the `flip` issue next.

**Review by DSGuala — APPROVED** [review]
Looks good!
